rest/data/generate.go

```diff
@@ -13,12 +14,34 @@ import (
 type GenerateConnector struct{}
 
 // GenerateTasks parses JSON files for `generate.tasks` and creates the new builds and tasks.
-func (gc *GenerateConnector) GenerateTasks(ctx context.Context, taskID string, jsonBytes []json.RawMessage, q amboy.Queue) error {
+func (gc *GenerateConnector) GenerateTasks(ctx context.Context, taskID string, jsonBytes []json.RawMessage, group amboy.QueueGroup) error {
+	t, err := task.FindOneId(taskID)
+	if err != nil {
+		return errors.Wrapf(err, "problem finding task %s", taskID)
+	}
+	q, err := group.Get(ctx, t.Version)
+	if err != nil {
+		return errors.Wrapf(err, "problem getting queue for version %s", t.Version)
+	}
 	return q.Put(units.NewGenerateTasksJob(taskID, jsonBytes))
 }
 
-func (gc *GenerateConnector) GeneratePoll(ctx context.Context, taskID string, queue amboy.Queue) (bool, []string, error) {
-	j, exists := queue.Get(fmt.Sprintf("generate-tasks-%s", taskID))
+func (gc *GenerateConnector) GeneratePoll(ctx context.Context, taskID string, group amboy.QueueGroup) (bool, []string, error) {
+	t, err := task.FindOneId(taskID)
+	if err != nil {
+		return false, nil, errors.Wrapf(err, "problem finding task %s", taskID)
+	}
+	q, err := group.Get(ctx, t.Version)
+	for stat := range q.JobStats(ctx) {
```

Comment:
left some debug code, though this might actually be useful to log?

Reply:
removed